File: solutions/c/pangram/1/pangram.c
#include <ctype.h>
#include <stddef.h>
#include <stdio.h>
#include <string.h>

#include "pangram.h"

bool is_pangram(char const *sentence) {
    if (sentence == NULL)
        return false;
    
    char c;
    int counts[26] = { 0 };
    while ((c = *sentence++)) {
        if (isalpha(c))
            counts[tolower(c) - 'a']++;
    }

    for (size_t i = 0; i < 26; ++i)
        if (!counts[i])
            return false;

    return true;
}

File: solutions/go/pangram/1/pangram.go
package pangram

import "strings"

func IsPangram(input string) bool {
	testString := strings.ToLower(input)
    for _, letter := range "abcdefghijklmnopqrstuvwxyz" {
        if !strings.ContainsRune(testString, letter) {
        	return false
        }
    }
    return true
}
